sql/angel-meneses.sql

```diff
@@ -0,0 +1,97 @@
+-- Initialize
+use coursera_db;
+go
+
+begin transaction;
+go
+
+
+-- 1. Añadir una nueva columna 'Edad' en la tabla Student. 
+-- Luego, llenar esa columna con las edades de los estudiantes cada vez que ellos registren su fecha de nacimiento en su información.
+-- Para aquellos que ya tienen registrada la fecha, actualizar el campo automáticamente
+
+ALTER TABLE Student ADD Age INT;
+go
+
+CREATE TRIGGER TX_SettingAge ON Student
+FOR UPDATE
+AS
+UPDATE Student SET Age = DATEDIFF(YEAR, birthday, CURRENT_TIMESTAMP);
+go
+
+UPDATE Student SET birthday = '2003-01-18' WHERE student_id = 3;
+UPDATE Student SET birthday = '2003-05-11' WHERE student_id = 5;
+UPDATE Student SET birthday = '2005-12-03' WHERE student_id = 7; 
+go
+
+-- 2. Coursera quiere saber quién fue el estudiante más joven que completó más cursos en un periodo de tiempo (entre 2 fechas)
+-- (función + subqueries)
+
+CREATE FUNCTION EstudianteCursosTerminados(@start DATE, @end DATE) RETURNS TABLE
+AS
+RETURN(SELECT S.fullname as "StudentName", S.Age , COUNT(M.course_id) AS "FinishedCourses"
+FROM Membership M INNER JOIN Student S ON  M.student_id = S.student_id
+WHERE M.finished_course = 1 AND M.enrollment_date BETWEEN @start AND @end AND S.birthday IS NOT NULL
+GROUP BY S.fullname, DATEDIFF(DAY, birthday, CURRENT_TIMESTAMP)
+HAVING COUNT(M.course_id) >= (SELECT MAX(ECT.FinishedCourses)
+FROM (SELECT S.fullname as "StudentName", COUNT(M.course_id) AS "FinishedCourses"
+FROM Membership M INNER JOIN Student S ON  M.student_id = S.student_id
+WHERE M.finished_course = 1 AND M.enrollment_date BETWEEN @start AND @end AND S.birthday IS NOT NULL
+GROUP BY S.fullname) ECT)
+AND DATEDIFF(DAY, birthday, CURRENT_TIMESTAMP) <= (SELECT MIN(ECT.AgeDays)
+FROM (SELECT S.fullname as "StudentName", DATEDIFF(DAY, birthday, CURRENT_TIMESTAMP) AS "AgeDays"
+FROM Membership M INNER JOIN Student S ON  M.student_id = S.student_id
+WHERE M.finished_course = 1 AND M.enrollment_date BETWEEN @start AND @end AND S.birthday IS NOT NULL) ECT))
+go;
```

Comment:
No lleva punto y coma la instrucción _go_. Por eso fallaron los _tests_. 


```suggestion
go
```